Common/Python/PandasData.cs

```diff
@@ -0,0 +1,280 @@
+
+                var tradeBar = bar as TradeBar;
+                if (tradeBar != null)
+                {
+                    _series["volume"].Add((double)tradeBar.Volume);
+                }
+
+                var quoteBar = bar as QuoteBar;
+                if (quoteBar != null)
+                {
+                    _series["asksize"].Add((double)quoteBar.LastAskSize);
+                    _series["bidsize"].Add((double)quoteBar.LastBidSize);
+
+                    if (quoteBar.Ask != null)
+                    {
+                        _series["askopen"].Add((double)quoteBar.Ask.Open);
+                        _series["askhigh"].Add((double)quoteBar.Ask.High);
+                        _series["asklow"].Add((double)quoteBar.Ask.Low);
+                        _series["askclose"].Add((double)quoteBar.Ask.Close);
+                    }
+                    else
+                    {
+                        _series["askopen"].Add(double.NaN);
+                        _series["askhigh"].Add(double.NaN);
+                        _series["asklow"].Add(double.NaN);
+                        _series["askclose"].Add(double.NaN);
+                    }
+
+                    if (quoteBar.Bid != null)
+                    {
+                        _series["bidopen"].Add((double)quoteBar.Bid.Open);
+                        _series["bidhigh"].Add((double)quoteBar.Bid.High);
+                        _series["bidlow"].Add((double)quoteBar.Bid.Low);
+                        _series["bidclose"].Add((double)quoteBar.Bid.Close);
+                    }
+                    else
+                    {
+                        _series["bidopen"].Add(double.NaN);
+                        _series["bidhigh"].Add(double.NaN);
+                        _series["bidlow"].Add(double.NaN);
+                        _series["bidclose"].Add(double.NaN);
+                    }
+                }
+            }
+
+            var tick = baseData as Tick;
+            if (tick != null)
+            {
+                if (tick.TickType == TickType.Quote)
+                {
+                    _series["askprice"].Add((double)tick.AskPrice);
+                    _series["asksize"].Add((double)tick.AskSize);
+                    _series["bidprice"].Add((double)tick.BidPrice);
+                    _series["bidsize"].Add((double)tick.BidSize);
+                }
+                else
+                {
+                    _series["exchange"].Add(tick.Exchange);
+                    _series["suspicious"].Add(tick.Suspicious);
+                }
+                _series["lastprice"].Add((double)tick.LastPrice);
+                _series["quantity"].Add((double)tick.Quantity);
+            }
+
+            var data = baseData as DynamicData;
+            if (data != null)
+            {
+                foreach (var kvp in data.GetStorageDictionary())
+                {
+                    _series[kvp.Key].Add(Convert.ToDouble(kvp.Value));
```

Comment:
`_series[kvp.Key]` is a `List<object>` no need to convert to double... maybe only convert if:
```csharp
var value = kvp.Value;
if (value is decimal) value = (double) value;
_series[kvp.Key].Add(value);
```